packages/nextjs/utils/scaffold-stark/contract.ts

```diff
@@ -386,10 +386,19 @@ export function getFunctionsByStateMutability(
 
 // TODO: in the future when param decoding is standarized in wallets argent and braavos we can return the object
 // TODO : starknet react makes an input validation so we need to return objects for function reads
-function tryParsingParamReturnValues(fn: (x: any) => {}, param: any) {
+// new starknet react hooks (v3) doesnt use raw parse
+function tryParsingParamReturnValues(
+  fn: (x: any) => {},
+  param: any,
+  isV3Parsing: boolean,
+) {
```

Comment:
We can remove above TODOs and write a note on why we use `isV3Parsing` flag here